maraboupy/test/test_nnet_extensions.py

```diff
@@ -0,0 +1,149 @@
+
+    nnet_object = MarabouNetworkNNet(filename=network_filename)
+    nnet_object.writeNNet(output_filename)
+
+    nnet_object_a = MarabouNetworkNNet(filename=output_filename)
+
+    N = 10
+    for i in range(N):
+        inputs = nnet_object_a.createRandomInputsForNetwork()
+
+        output1 = nnet_object.evaluateNetwork(inputs,normalize_inputs=False,normalize_outputs=False)
+        output2 = nnet_object_a.evaluateNetwork(inputs,normalize_inputs=False,normalize_outputs=False)
+        assert (output1 == output2).all()
+
+        # Compare evaluation with and without Marabou
+        without_marabou_output = nnet_object_a.evaluate(np.array([inputs]),useMarabou=False).flatten()
+        without_marabou_output_rounded = np.array([float(round(y,8)) for y in without_marabou_output])
+
+        with_marabou_output = nnet_object_a.evaluate(np.array([inputs]),useMarabou=True).flatten()
+        with_marabou_output_rounded = np.array([float(round(y,8)) for y in with_marabou_output])
+
+        # Assert that all of the above agree, at least up to 10^-8
+        assert (output2 == without_marabou_output).all()
+        assert (without_marabou_output_rounded == with_marabou_output_rounded).all()
+
+        # Adding input and output normalization
+
+        output1 = nnet_object.evaluateNetwork(inputs, normalize_inputs=True, normalize_outputs=True)
+        output2 = nnet_object_a.evaluateNetwork(inputs, normalize_inputs=True, normalize_outputs=True)
+        assert (output1 == output2).all()
+
+
+
+def test_normalize_read_flag():
+    '''
+    Similar tests to those in test_write_read_evaluate(), but turns normalize flag on when creating the
+    MarabouNetworkNNet objects.
+    Note that in this case inputs and outputs need to be normalized.
+    '''
+    output_filename = "./maraboupy/test/ACASXU_experimental_v2a_1_9_output.nnet"
+
+    nnet_object = MarabouNetworkNNet(filename=network_filename, normalize=True)
+    nnet_object.writeNNet(output_filename)
+
+    nnet_object_a = MarabouNetworkNNet(filename=output_filename, normalize=True)
+
+    N = 10
+    for i in range(N):
+        inputs = nnet_object_a.createRandomInputsForNetwork()
+
+        output1 = nnet_object.evaluateNetwork(inputs, normalize_inputs=True, normalize_outputs=True)
+        output2 = nnet_object_a.evaluateNetwork(inputs,normalize_inputs=True, normalize_outputs=True)
+        assert (output1 == output2).all()
+
+        without_marabou_output = nnet_object_a.evaluate(np.array([inputs]),useMarabou=False).flatten()
+        assert (output2 == without_marabou_output).all()
+
+        with_marabou_output = nnet_object_a.evaluate(np.array([inputs]), useMarabou=True).flatten()
+        with_marabou_output_rounded = np.array([float(round(y, 8)) for y in with_marabou_output])
+        output2_rounded = np.array([float(round(y, 8)) for y in output2])
+        assert (output2_rounded == with_marabou_output_rounded).all()
+
+
+if __name__ == "__main__":
```

Comment:
For consistency, it would be good to follow the same type of format as the other test files and use pytest, in which case this main function is not needed.